File: src/jsonrpc.ts
import type { RpcRequest, RpcResponse } from './types.ts';

export function isRpcRequest(value: unknown): value is RpcRequest {
  return !!value && typeof value === 'object' && 'method' in value;
}

export function rpcError(id: RpcRequest['id'], code: number, message: string): RpcResponse {
  return { jsonrpc: '2.0', id: id ?? null, error: { code, message } };
}

export function rpcResult(id: RpcRequest['id'], result: unknown): RpcResponse {
  return { jsonrpc: '2.0', id: id ?? null, result };
}

File: .code_index/skeleton/src/jsonrpc.py
from __future__ import annotations

from .types import RpcRequest, RpcResponse

# @origin src/jsonrpc.ts:3
def isRpcRequest(value: Any) -> value is RpcRequest:
    ...

# @origin src/jsonrpc.ts:7
def rpcError(id: RpcRequest[id], code: float, message: str) -> RpcResponse:
    ...

# @origin src/jsonrpc.ts:11
def rpcResult(id: RpcRequest[id], result: Any) -> RpcResponse:
    ...
